controllers/etcd_controller.go

```diff
@@ -251,6 +271,42 @@ func (r *EtcdReconciler) reconcile(ctx context.Context, etcd *druidv1alpha1.Etcd
 func (r *EtcdReconciler) delete(ctx context.Context, etcd *druidv1alpha1.Etcd) (ctrl.Result, error) {
 	logger.Infof("Deletion timestamp set for etcd: %s", etcd.GetName())
 
+	svc, err := r.getService(ctx, etcd)
+	if err != nil {
+		logger.Error(err, "error getting service")
+		return ctrl.Result{
+			Requeue: true,
+		}, err
+	}
+
+	ss, err := r.getStatefulSet(ctx, etcd)
```

Comment:
Can we find a way to modularize the code to avoid repetition of [this](https://github.com/gardener/etcd-druid/pull/59/files#diff-8e5c816f8b2f1adeef36a64d81b3e0f6R218-R234) here?

Reply:
Since error handling for getting sts and listing PVCs is different (`Requeue` and `RequeueAfter`), combining the two operations into a single method would require a new switch/if condition or some extra code to handle the common error from the method, which might be unnecessary.

Reply:
I was thinking, maybe the common mention might return the `ctrl.Result` in addition to error and then in the reconcile/delete functions we could return whatever result was returned in case of error. To use `Requeue` or `RequeueAfter` is dependent on the statefulset or PVCs and not on whether it is called from reconcile or delete.